lib/Driver/ToolChains.cpp

```diff
@@ -1044,6 +1044,24 @@ addLinkSanitizerLibArgsForDarwin(const ArgList &Args,
                      /*AddRPath*/ true, TC);
 }
 
+static void
+addLinkFuzzerLibArgsForDarwin(const ArgList &Args,
+                       ArgStringList &Arguments,
+                       const ToolChain &TC) {
+
+  // libFuzzer requires C++.
+  Arguments.push_back("-lc++");
+
+  // Link libfuzzer.
+  SmallString<128> Dir;
+  getRuntimeLibraryPath(Dir, Args, TC);
+  llvm::sys::path::remove_filename(Dir);
+  llvm::sys::path::append(Dir, "llvm", "libLLVMFuzzer.a");
```

Comment:
I'm not sure what you mean by "all kind of libraries shipped with LLVM". If you mean the lib/ directory inside the build directory, that's not *installed* along with Clang or Swift, and so the Swift compiler must not reference things in it (even by symlink).

For testing purposes I think you should just pass the library explicitly for now, like people do for Clang. We can put it into either the Clang or Swift resource dirs when we're ready to ship this feature.